src/hasAtLeast.ts

```diff
@@ -1,14 +1,41 @@
-import type { ReadonlyTuple } from "./_types";
+import type { IterableContainer } from "./_types";
 import { purry } from "./purry";
 
-type ArrayMinN<T, N extends number> = number extends N
-  ? // We can only compute the type for a literal number!
-    Array<T>
-  : // I don't know why we need to special-case the 0 case, but otherwise
-    // typescript complains we have a deep recursion.
-    N extends 0
-    ? Array<T>
-    : [...ReadonlyTuple<T, N>, ...Array<T>];
+type ArraySetRequired<
```

Comment:
**question:** should we file an issue with type-fest?

Reply:
We used a copy-pasted version from a few months ago; I didn't check back to see if they changed their implementation for ts5.4. Re-reading our code, I thought that building our own version made sense here.

Generally, my feelings towards type-fest shifted with the work for v2, as many type-fest types don't make sense in real-world use cases, feel more "academic"/"challenge-like"  than practical, and aren't always helpful when typing our functions. I'd rather reduce our surface area to where we use their types as-is when they fit perfectly (or we adjust the runtime behavior to match their type).